tests/test_smoke.py

```diff
@@ -2434,6 +2446,38 @@ def test_new_bucket_creation_and_deletion(self, tmp_local_storage_obj,
         out = subprocess.check_output(['sky', 'storage', 'ls'])
         assert tmp_local_storage_obj.name not in out.decode('utf-8')
 
+    @pytest.mark.parametrize('store_type', [
+        storage_lib.StoreType.S3, storage_lib.StoreType.GCS,
+        pytest.param(storage_lib.StoreType.R2, marks=pytest.mark.cloudflare)
+    ])
```

Comment:
This test seems to be failing transiently for GCS:

```
(base) ➜  sky-experiments git:(storage_delete_parallel) ✗ pytest tests/test_smoke.py::TestStorageWithCredentials::test_multiple_buckets_creation_and_deletion
bringing up nodes...
s.F
=================================================================================================================================================================================== FAILURES ===================================================================================================================================================================================
____________________________________________________________________________________________________________________________________________ TestStorageWithCredentials.test_multiple_buckets_creation_and_deletion[StoreType.GCS] _____________________________________________________________________________________________________________________________________________
[gw0] darwin -- Python 3.9.13 /Users/romilb/tools/anaconda3/bin/python
tests/test_smoke.py:2476: in test_multiple_buckets_creation_and_deletion
    assert exists, (f"Storage not created. Expected: {storage_obj_name} "
E   AssertionError: Storage not created. Expected: ['sky-test-1687466361968251', 'sky-test-16874663619683762', 'sky-test-1687466361968394', 'sky-test-1687466361968401', 'sky-test-16874663619684072'] Actual: NAME
E     sky-test-1687466361968394
E     sky-test-1687466361968401
E     sky-test-16874663619684072
E     
E   assert False
=========================================================================================================================================================================== short test summary info ============================================================================================================================================================================
FAILED tests/test_smoke.py::TestStorageWithCredentials::test_multiple_buckets_creation_and_deletion[StoreType.GCS] - AssertionError: Storage not created. Expected: ['sky-test-1687466361968251', 'sky-test-16874663619683762', 'sky-test-1687466361968394', 'sky-test-1687466361968401', 'sky-test-16874663619684072'] Actual: NAME
1 failed, 1 passed, 1 skipped, 1197 warnings in 15.91s
```

Might be something to do with S3 and GCS tests running in parallel...

(Following up on email to grant GCP access for testing this)

Reply:
This test is updated with a fixture as suggested and also with dependencies for store types so that it can run for S3, GCS and R2 without failures. Previously the failure was due to concurrent execution of `sky storage delete -a` for all store types (S3, GCS and R2)